lib/widgets/path_editor/cards/graph_settings_card.dart

```diff
@@ -0,0 +1,295 @@
+            bool state =
+                widget.prefs.getBool(GraphEditor.prefShowVelocity) ?? true;
+            widget.prefs.setBool(GraphEditor.prefShowVelocity, !state);
+            widget.onUpdate();
+          },
+        ),
+        Text(
+          'Velocity',
+          style: TextStyle(color: colorScheme.onSurface),
+        ),
+        const SizedBox(width: 12),
+        Checkbox(
+          value: widget.prefs.getBool(GraphEditor.prefShowAccel),
+          activeColor: GraphEditor.colorAccel,
+          checkColor: GraphEditor.colorAccel,
+          side: BorderSide(color: GraphEditor.colorAccel, width: 2),
+          onChanged: (val) {
+            bool state =
+                widget.prefs.getBool(GraphEditor.prefShowAccel) ?? true;
+            widget.prefs.setBool(GraphEditor.prefShowAccel, !state);
+            widget.onUpdate();
+          },
+        ),
+        Text(
+          'Acceleration',
+          style: TextStyle(color: colorScheme.onSurface),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildHeadingAngularRow(BuildContext context) {
+    ColorScheme colorScheme = Theme.of(context).colorScheme;
+
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      mainAxisSize: MainAxisSize.max,
+      children: [
+        Checkbox(
+          value: widget.prefs.getBool(GraphEditor.prefShowHeading),
+          activeColor: GraphEditor.colorHeading,
+          checkColor: GraphEditor.colorHeading,
+          side: BorderSide(color: GraphEditor.colorHeading, width: 2),
+          onChanged: (val) {
+            bool state =
+                widget.prefs.getBool(GraphEditor.prefShowHeading) ?? true;
+            widget.prefs.setBool(GraphEditor.prefShowHeading, !state);
+            widget.onUpdate();
+          },
+        ),
+        Text(
+          'Heading',
+          style: TextStyle(color: colorScheme.onSurface),
+        ),
+        const SizedBox(width: 12),
+        Checkbox(
+          value: widget.prefs.getBool(GraphEditor.prefShowAngularVelocity),
+          activeColor: GraphEditor.colorAngularVelocity,
+          checkColor: GraphEditor.colorAngularVelocity,
+          side: BorderSide(color: GraphEditor.colorAngularVelocity, width: 2),
+          onChanged: (val) {
+            bool state =
+                widget.prefs.getBool(GraphEditor.prefShowAngularVelocity) ??
+                    true;
+            widget.prefs.setBool(GraphEditor.prefShowAngularVelocity, !state);
+            widget.onUpdate();
+          },
+        ),
+        Text(
+          'Angular',
+          style: TextStyle(color: colorScheme.onSurface),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildCurvatureRow(BuildContext context) {
+    ColorScheme colorScheme = Theme.of(context).colorScheme;
+
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      mainAxisSize: MainAxisSize.max,
+      children: [
+        Checkbox(
+          value: widget.prefs.getBool(GraphEditor.prefShowCurvature),
```

Comment:
OK, I can see how this might make it crash for you. I have those preferences set through numerous iterations to get from nothing to what you see now...so I've ended up with something that "works for me" as a result. I've looked around (for maybe 10 minutes...with no luck) trying to figure out how to wipe out all the PathPlanner preferences from my computer so I can run it from a fresh system. I was hoping to be able to reproduce the crash you see, and therefore confirm that the next set of changes will fix that crash. What you propose (and I've done) _seems_ like it should do the trick. Knowing instead of guessing would be so much better! ;-)

Reply:
Well the cause is from not having those preferences saved previously so the default values in initState will fix it but if you wanted to make extra sure you can delete shared_preferences.json in `%AppData%\com.mjansen4857\pathplanner` assuming you are on windows. If you're on mac its more trouble than its worth.

Reply:
You could do it, you just have to delete preferences individually from the command line. There might be a way to delete all of them but I'm unsure an example would be: `defaults delete com.mjansen4857.pathplanner flutter.showVelocity`

Although after looking at the preference keys you have set, I'd probably change them to have a prefix of "graph" or something similar so they aren't so generic. Changing the names would have the same effect of deleting all of them.

Reply:
Good general suggestion on the setting prefix. After adding it, the app crashed for me as well. After the initState() fixes, it no longer crashes. Yeah!

Reply:
Oh, `defaults delete com.mjansen4857.pathplanner` will delete all the settings for PathPlanner. Simple!

Reply:
Ah, ok. Didn't want to try that myself incase it messes something else up in the defaults that sharedpreferences doesn't manage.

Reply:
Ok I'll look into that now. I did that so long ago I don't remember either. That might explain a crash someone else was having earlier... oops.